File: atcoder/go/abc154/d.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	r := bufio.NewReader(os.Stdin)
	var n, k int
	fmt.Fscan(r, &n, &k)
	p := make([]float64, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(r, &p[i])
	}
	var ans float64 = -1
	var s float64 = 0
	for i := 0; i < k; i++ {
		s += (1 + p[i]) / 2.0
	}
	ans = max(ans, s)
	for i := k; i < n; i++ {
		s += (p[i]+1)/2.0 - (p[i-k]+1)/2.0
		ans = max(ans, s)
	}
	fmt.Println(ans)
}
func max(a, b float64) float64 {
	if a > b {
		return a
	} else {
		return b
	}
}

File: atcoder/python/abc154/d.py
n,k=map(int,input().split())
p=list(map(int,input().split()))
s=0
for i in range(k):
    s+=(p[i]+1)/2
ans=s
for i in range(k,n):
    s+=(p[i]+1)/2-(p[i-k]+1)/2
    ans=max(ans,s)
print(ans)